bin/module-update/module-update.sh

```diff
@@ -0,0 +1,185 @@
+BACKUP_DIR="${DEPLOY_DIR}-BACKUP-${TIMESTAMP}"
+
+[ -f "$META_PATH" ] && source "$META_PATH"
+
+update_hosted_zone_configuration() {
+
+  printf "Hosted zone setup has been moved to IRSA."
+
+  local hosted_zone=$(hcledit attribute get route53_hosted_zone_name -f "${INFRA_VARS}")
+  local irsa_external_dns=$(hcledit attribute get "irsa_external_dns" -f "${CLUSTER_VARS}")
+
+  if [ -n "$hosted_zone" ]; then
+    if [ -z "$irsa_external_dns" ]; then
+      cat <<-EOB >>"$CLUSTER_VARS"
+irsa_external_dns = {
+  enabled = true
+  hosted_zone_name = $hosted_zone
+}
+EOB
+    else
+      printf "'irsa_external_dns' already exists on %s.\n" "${CLUSTER_VARS}"
+    fi
+  fi
+
+  printf "Removing route53_hosted_zone_private var.\n"
+
+  hcledit block rm variable.route53_hosted_zone_name -u -f "${INFRA_DIR}/variables.tf"
+  hcledit block rm variable.route53_hosted_zone_private -u -f "${INFRA_DIR}/variables.tf"
+
+  hcledit attribute rm module.infra.route53_hosted_zone_name -u -f "${INFRA_DIR}/main.tf"
+  hcledit attribute rm module.infra.route53_hosted_zone_private -u -f "${INFRA_DIR}/main.tf"
+
+  hcledit attribute rm route53_hosted_zone_name -u -f "${INFRA_VARS}"
+
+}
+
+set_vpc_cni_import() {
+
+  local nodes_main_tf_path="${NODES_DIR}/imports.tf"
+
+  if hcledit block get "import" -f "$nodes_main_tf_path" | grep -q 'to = module.nodes.aws_eks_addon.pre_compute_addons\["vpc-cni"\]'; then
+    printf 'Import block already exists for module.nodes.aws_eks_addon.pre_compute_addons["vpc-cni"]. \n'
+    return
+  fi
+
+  printf "Creating import block for 'vpc-cni' EKS addon. After this configuration is applied and the vpc-cni addon is imported, the %s file can be removed\n" "$nodes_main_tf_path"
+
+  cat <<-EOB >>"$nodes_main_tf_path"
+import {
+  to = module.nodes.aws_eks_addon.pre_compute_addons["vpc-cni"]
+  id = "\${local.eks.cluster.specs.name}:vpc-cni"
+}
+EOB
+
+  echo
+}
+
+update_mod_source() {
+
+  local mod_dir="$1"
+  local mod_version="$2"
+
+  local update_mod_sh_path="${mod_dir}/set-mod-version.sh"
+
+  if [ ! -f "$update_mod_sh_path" ]; then
+    printf "Unable to find %s\nExiting...\n" "$update_mod_sh_path"
+    exit 1
+  fi
+
+  "$update_mod_sh_path" "$mod_version"
+}
+
+initialize_upgrade_module() {
+  if [ -z "$UPGRADE_DIR" ] || [ -z "$MOD_VERSION" ]; then
+    printf "UPGRADE_DIR and MOD_VERSION vars need to be set.\nExiting...\n"
+    exit 1
+  fi
+
+  printf "Creating Upgrade deploy DIR..."
+  mkdir -p "$UPGRADE_DIR"
+
+  printf "Initializing module on %s\n" "$UPGRADE_DIR"
+  terraform -chdir="$UPGRADE_DIR" init -backend=false -from-module="github.com/dominodatalab/terraform-aws-eks.git//examples/deploy?ref=${MOD_VERSION}"
+
+  printf "Updating module source...\n"
+  update_mod_source "$UPGRADE_DIR" "$MOD_VERSION"
+}
+
+backup_files() {
+  if [ ! -d "$BACKUP_DIR" ] || [ -z "$(ls -A "$BACKUP_DIR")" ]; then
+    printf "Backing up %s to %s.\n" "$DEPLOY_DIR" "$BACKUP_DIR"
+    cp -r "$DEPLOY_DIR" "$BACKUP_DIR"
```

Comment:
Would be good to do trailing slash here so you don't end up with `$BACKUP_DIR/$DEPLOY_DIR` if this gets retried.